api/lib/domain/services/badge-criteria-service.js

```diff
@@ -13,10 +14,16 @@ function areBadgeCriteriaFulfilled({ campaignParticipationResult, badgeCriteria
         );
         break;
       case BadgeCriterion.SCOPES.EVERY_PARTNER_COMPETENCE :
-        isBadgeCriterionFulfilled = _verifyEveryPartnerCompetenceResultMasteryPercentageCriterion(
-          campaignParticipationResult.partnerCompetenceResults,
-          criterion.threshold
+        campaignParticipationBadge = _.find(
+          campaignParticipationResult.campaignParticipationBadges,
+          (campaignParticipationBadge) => campaignParticipationBadge.id === badge.id
         );
+        if (campaignParticipationBadge) {
+          isBadgeCriterionFulfilled = _verifyEveryPartnerCompetenceResultMasteryPercentageCriterion(
+            campaignParticipationBadge.partnerCompetenceResults,
+            criterion.threshold
+          );
+        }
```

Comment:
Avec le commentaire du dessus, on obtient : 

```suggestion
          isBadgeCriterionFulfilled = _verifyEveryPartnerCompetenceResultMasteryPercentageCriterion(
            campaignParticipationBadge.partnerCompetenceResults,
            criterion.threshold
          );
```

Reply:
Je n'ai pas bien saisi la suggestion, mais j'ai changé l'implémentation de la fonction, si tu peux y jeter un oeil =)